pkg/gadget-context/gadget-context.go

```diff
@@ -65,41 +72,63 @@ func New(
 		logger:                   logger,
 		operators:                operators.GetOperatorsForGadget(gadget),
 		operatorsParamCollection: operatorsParamCollection,
+		timeout:                  timeout,
 	}
 }
 
-func (r *GadgetContext) ID() string {
-	return r.id
+func (c *GadgetContext) ID() string {
+	return c.id
+}
+
+func (c *GadgetContext) Context() context.Context {
+	return c.ctx
+}
+
+func (c *GadgetContext) Cancel() {
+	c.cancel()
 }
 
-func (r *GadgetContext) Context() context.Context {
-	return r.ctx
+func (c *GadgetContext) Parser() parser.Parser {
+	return c.parser
 }
 
-func (r *GadgetContext) Parser() parser.Parser {
-	return r.parser
+func (c *GadgetContext) Runtime() runtime.Runtime {
+	return c.runtime
 }
 
-func (r *GadgetContext) Runtime() runtime.Runtime {
-	return r.runtime
+func (c *GadgetContext) GadgetDesc() gadgets.GadgetDesc {
+	return c.gadget
 }
 
-func (r *GadgetContext) GadgetDesc() gadgets.GadgetDesc {
-	return r.gadget
+func (c *GadgetContext) Operators() operators.Operators {
+	return c.operators
 }
 
-func (r *GadgetContext) Operators() operators.Operators {
-	return r.operators
+func (c *GadgetContext) Logger() logger.Logger {
+	return c.logger
 }
 
-func (r *GadgetContext) Logger() logger.Logger {
-	return r.logger
+func (c *GadgetContext) GadgetParams() *params.Params {
+	return c.gadgetParams
 }
 
-func (r *GadgetContext) GadgetParams() *params.Params {
-	return r.gadgetParams
+func (c *GadgetContext) OperatorsParamCollection() params.Collection {
+	return c.operatorsParamCollection
 }
 
-func (r *GadgetContext) OperatorsParamCollection() params.Collection {
-	return r.operatorsParamCollection
+func (c *GadgetContext) Timeout() time.Duration {
+	return c.timeout
+}
+
+func (c *GadgetContext) WaitForTimeoutOrDone() {
+	ctx, cancel := WithTimeoutOrCancel(c.ctx, c.timeout)
+	defer cancel()
+	<-ctx.Done()
+}
```

Comment:
I have mixed feelings about this function. I know it's useful in many cases, however it seems that it's too specific to be in   `GadgetContext`. I was wondering if this could be a helper function like, so we don't have to add this method to interfaces around. 

```golang
func WaitForTimeoutOrDone(c gadgets.GadgetContext) {
	ctx, cancel := WithTimeoutOrCancel(c.Context(), c.Timeout())
	defer cancel()
	<-ctx.Done()
}
```

and gadgets will use it like:

```golang 
	gadgetcontext.WaitForTimeoutOrDone(gadgetCtx)
```

but I don't know if that's a good idea. 

Also, this function won't work in cases like https://github.com/inspektor-gadget/inspektor-gadget/pull/1408#discussion_r1136158925

Reply:
TBH, I also had the same mixed feelings. And not that you mentioned it, I will make it a helper function.